### DISCUSSION TITLE
"Should transform.totensor be used, and when should it be used?"

### DISCUSSION THREAD

**Discussion by hz1z** [Project-MONAI/MONAI#8008] [General]
![IMG_20240809_181051.jpg](https://github.com/user-attachments/assets/0a51df63-ddb0-43b5-9fbd-a5b887f7f280)

I saw someone's blog writing it this way, but I noticed that the official tutorial does not use transform.totensord. At first, I thought transform.totensord would convert my data from MetaTensor type to PyTorch's tensor type to save memory. However, I found that whether or not I use transform.totensord in the compose, the data ends up being MetaTensor type. Should I use transform.totensord? And when should it be used?"

**Comment by KumoLiu** [comment]
Hi @hz1z, thanks for your interest here.
If you want to convert your data to Tensor without meta information, you can do something like this:
```
ToTensord(track_meta=False)
```
Hope it helps, thanks.

**Comment by hz1z** [comment]
@KumoLiu  I don’t want to convert metatensor to tensor.
Do you think it’s unnecessary to use totensor in his blog post?
 I have two questions that I hope you can answer: 
1.  Do you recommend using transform.totensor in the compose? 
2.  If you would not use transform.totensor in the compose, then what is the purpose of this function? Thank you!!!

**Reply by KumoLiu** [reply]
Use it when you want to converts the input image to a tensor without applying any other transformations. If your dataset is already metatensor and you don't want to convert it to tensor, then no need to add this transform.
Thanks.
